machine/v1beta1/types_azureprovider.go

```diff
@@ -145,6 +145,11 @@ type AzureMachineProviderSpec struct {
 	// This is useful for debugging software based launch issues.
 	// +optional
 	Diagnostics AzureDiagnostics `json:"diagnostics,omitempty"`
+	// CapacityReservationGroupID specifies the capacity reservation group resource id that should be
+	// used for allocating the virtual machine.
+	// +kubebuilder:validation:XValidation:rule="oldSelf == '' || self == oldSelf",message="CapacityReservationGroupID is immutable once set"
+	// +optional
+	CapacityReservationGroupID string `json:"capacityReservationGroupID,omitempty"`
```

Comment:
What is the format of the reservation group ID? What characters are valid? What is the minimal and maximal length?

Reply:
@JoelSpeed I found this is the format for Capacity reservation Group ID 

I guess resourceGroupName and CapacityReservationGroupName may vary on differen Users.
`subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/capacityReservationGroups/<CapacityReservationGroupName>`

Also didn't found maximum and minimum length for ID in AzureDocs.

Ref: https://learn.microsoft.com/en-us/azure/virtual-machines/capacity-reservation-create?tabs=cli1%2Capi1%2Capi2#create-a-capacity-reservation-1

Reply:
There's also a validation that it is a valid azure ID no?

Reply:
Yes, there  is a validation that the input must be string  having length more than 0,
Also the string should start with "/"

The keys which are used should be amoung "subscriptions", "providers" and "resourcegroups"
following by valid IDs respectively. 

the ideal format for ID is `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/capacityReservationGroups/<CapacityReservationGroupName>`